File: network/src/main/java/com/mooc/network/ApiResponse.java
package com.mooc.network;

public class ApiResponse<T> {
    public static final int CACHE_CODE = 304;
    public static final int LOCAL_ERROR = 999;

    public int status;
    public String message;
    public T data;


    public boolean isSuccessful() {
        return status >= 200 && status < 300;
    }

    @Override
    public String toString() {
        return "ApiResponse{" +
                "status=" + status +
                ", message='" + message + '\'' +
                ", data=" + data +
                '}';
    }

    public boolean isCached() {
        return status == CACHE_CODE;
    }
}

File: network/src/main/java/com/mooc/network/ApiResponse.kt
package com.mooc.network

const val CACHE_CODE = 304
const val LOCAL_ERROR = 999

data class ApiResponse<T>(var status: Int, var message: String? = null, var data: T? = null)

val ApiResponse<*>.isSuccessful: Boolean
    get() = status in 200 until 300

val ApiResponse<*>.isCached: Boolean
    get() = status == CACHE_CODE